src/legacy/core_plugins/kibana/public/dashboard/panel/dashboard_panel.tsx

```diff
@@ -17,127 +17,162 @@
  * under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import { injectI18n, FormattedMessage } from '@kbn/i18n/react';
+import { FormattedMessage, injectI18n } from '@kbn/i18n/react';
+
+import { EuiPanel } from '@elastic/eui';
 import classNames from 'classnames';
 import _ from 'lodash';
-
-import { PanelHeader } from './panel_header';
+import React from 'react';
+import {
+  ContainerState,
+  Embeddable,
+  EmbeddableFactory,
+  EmbeddableMetadata,
+  EmbeddableState,
+} from 'ui/embeddable';
+import { EmbeddableActions, EmbeddableErrorAction } from '../actions';
+import { PanelState } from '../selectors';
 import { PanelError } from './panel_error';
+import { PanelHeader } from './panel_header/panel_header';
+
+interface DashboardPanelProps {
+  intl: any;
+  viewOnlyMode: boolean;
+  onPanelFocused: (panel: string) => {};
+  onPanelBlurred: (panel: string) => {};
+  error: string | object;
```

Comment:
```suggestion
   error?: string | object;
```

Reply:
Changing this to optional results in the following errors that were then resolved when marking 
`    error?: string | object;` in line 150 as well

> 
> 	"message": "Argument of type 'typeof DashboardPanelUi' is not assignable to parameter of type 'ComponentType<DashboardPanelProps>'.\n  Type 'typeof DashboardPanelUi' is not assignable to type 'ComponentClass<DashboardPanelProps>'.\n    Type 'DashboardPanelUi' is not assignable to type 'Component<DashboardPanelProps, ComponentState, any>'.\n      Types of property 'shouldComponentUpdate' are incompatible.\n        Type '(nextProps: { containerState: ContainerState; error: string | object; initialized: boolean; }) => boolean' is not assignable to type '(nextProps: Readonly<DashboardPanelProps>, nextState: Readonly<ComponentState>, nextContext: any) => boolean'.\n          Types of parameters 'nextProps' and 'nextProps' are incompatible.\n            Type 'Readonly<DashboardPanelProps>' is not assignable to type '{ containerState: ContainerState; error: string | object; initialized: boolean; }'.",
> 	"source": "ts",
> 	"startLineNumber": 198,
> 	"startColumn": 42,
> 	"endLineNumber": 198,
> 	"endColumn": 58
> }

Reply:
You should actually use the same interface for `nextProps`, it now has conflicting types which is causing the error. so instead of
```
nextProps: {
     containerState: ContainerState;
     error: string | object;
     initialized: boolean;
   })
```

do
```
nextProps: DashboardPanelProps)
```